xapi-libs and xenstore local customization for XenServer

The XenServer specific config and customization to the common libs and Xenstored.

Note that these are not appropriate for upstreamming and should stay in our patch queue. Any future updates to commonly shared version of xapi-libs/xenstore should go to the other patch xapi-libs-update to be upstreamed from time to time.

Signed-off-by: Zheng Li <zheng.li@eu.citrix.com>

diff --git a/tools/ocaml/xenstored/define.ml b/tools/ocaml/xenstored/define.ml
index 89a6aac..512a21f 100644
--- a/tools/ocaml/xenstored/define.ml
+++ b/tools/ocaml/xenstored/define.ml
@@ -25,8 +25,8 @@ let xs_daemon_socket_ro = "/var/run/xenstored/socket_ro"
 
 let default_config_dir = "/etc/xen"
 
-let maxwatch = ref (50)
-let maxtransaction = ref (20)
+let maxwatch = ref (-1)
+let maxtransaction = ref (0)
 
 let domid_self = 0x7FF0
 
diff --git a/tools/ocaml/xenstored/oxenstored.conf b/tools/ocaml/xenstored/oxenstored.conf
index dd20eda..e453550 100644
--- a/tools/ocaml/xenstored/oxenstored.conf
+++ b/tools/ocaml/xenstored/oxenstored.conf
@@ -31,7 +31,8 @@ persistent = false
 # access-log-file = /var/log/xenstored-access.log
 # access-log-nb-lines = 13215
 # acesss-log-nb-chars = 180
-# access-log-special-ops = false
+access-log-special-ops = false
+access-log-file = syslog:local3
 
 # Perodically scanning all the rings as a safenet for lazy clients.
 # Define the interval in seconds, set to negative to disable.
diff --git a/tools/ocaml/xenstored/quota.ml b/tools/ocaml/xenstored/quota.ml
index e6953c6..d34a9d7 100644
--- a/tools/ocaml/xenstored/quota.ml
+++ b/tools/ocaml/xenstored/quota.ml
@@ -19,9 +19,9 @@ exception Data_too_big
 exception Transaction_opened
 
 let warn fmt = Logging.warn "quota" fmt
-let activate = ref true
-let maxent = ref (10000)
-let maxsize = ref (4096)
+let activate = ref false
+let maxent = ref (-1)
+let maxsize = ref (-1)
 
 type t = {
 	maxent: int;               (* max entities per domU *)
diff --git a/tools/ocaml/xenstored/transaction.ml b/tools/ocaml/xenstored/transaction.ml
index 77de4e8..2647b0d 100644
--- a/tools/ocaml/xenstored/transaction.ml
+++ b/tools/ocaml/xenstored/transaction.ml
@@ -18,7 +18,7 @@ open Stdext
 
 let none = 0
 let test_eagain = ref false
-let do_coalesce = ref true
+let do_coalesce = ref false
 
 let check_parents_perms_identical root1 root2 path =
 	let hierarch = Store.Path.get_hierarchy path in
